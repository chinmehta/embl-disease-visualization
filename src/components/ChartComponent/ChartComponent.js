import React from "react";
import { Bar, Radar } from "react-chartjs-2";
import {
  Chart,
  ArcElement,
  LineElement,
  BarElement,
  PointElement,
  BarController,
  BubbleController,
  DoughnutController,
  LineController,
  PieController,
  PolarAreaController,
  RadarController,
  ScatterController,
  CategoryScale,
  LinearScale,
  LogarithmicScale,
  RadialLinearScale,
  TimeScale,
  TimeSeriesScale,
  Decimation,
  Filler,
  Legend,
  Title,
  Tooltip,
} from "chart.js";

Chart.register(
  ArcElement,
  LineElement,
  BarElement,
  PointElement,
  BarController,
  BubbleController,
  DoughnutController,
  LineController,
  PieController,
  PolarAreaController,
  RadarController,
  ScatterController,
  CategoryScale,
  LinearScale,
  LogarithmicScale,
  RadialLinearScale,
  TimeScale,
  TimeSeriesScale,
  Decimation,
  Filler,
  Legend,
  Title,
  Tooltip
);

function ChartComponent({ type, data }) {
  let chartType;
  const EMBL_PRIMARY_BLUE = "#3489ca";
  let state = {
    labels: ["january", "February", "March", "April", "May"],
    datasets: [
      {
        label: "Rainfall",
        data: [65, 59, 80, 81, 56],
        backgroundColor: EMBL_PRIMARY_BLUE,
        borderColor: EMBL_PRIMARY_BLUE,
      },
    ],
  };
  if (type === "bar") {
    chartType = <Bar className="max-h-full max-w-full" data={state} />;
  } else if (type === "radar") {
    state.datasets[0].backgroundColor = "transparent";
    chartType = <Radar className="max-h-full max-w-full" data={state} />;
  }
  return <React.Fragment>{chartType}</React.Fragment>;
}

export default ChartComponent;
