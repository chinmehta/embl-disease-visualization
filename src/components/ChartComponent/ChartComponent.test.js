import React from "react";
import renderer from "react-test-renderer";
import ChartComponent from "./ChartComponent";

it("renders when no data and type is passed as prop", () => {
  const element = renderer.create(<ChartComponent />).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when data and type is passed as prop -radar", () => {
  const data = {
    label: ["label 1", "label 2", "label 3"],
    chartDataArray: [0.3, 0.2, 0.9],
    approvedSymbol: "EMB",
  };
  const element = renderer
    .create(<ChartComponent type="radar" data={data} />)
    .toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when data and type is passed as prop -bar", () => {
  const data = {
    label: ["label 1", "label 2", "label 3"],
    chartDataArray: [0.3, 0.2, 0.9],
    approvedSymbol: "EMB",
  };
  const element = renderer
    .create(<ChartComponent type="bar" data={data} />)
    .toJSON();

  expect(element).toMatchSnapshot();
});
