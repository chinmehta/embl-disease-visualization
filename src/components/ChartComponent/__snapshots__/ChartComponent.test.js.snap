// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when data and type is passed as prop -bar 1`] = `
<canvas
  className="max-h-full max-w-full"
  height={150}
  role="img"
  width={300}
/>
`;

exports[`renders when data and type is passed as prop -radar 1`] = `
<canvas
  className="max-h-full max-w-full"
  height={150}
  role="img"
  width={300}
/>
`;

exports[`renders when no data and type is passed as prop 1`] = `
<canvas
  className="max-h-full max-w-full"
  height={150}
  role="img"
  width={300}
/>
`;
